spec/js_routes/zzz_last_post_rails_init_spec.rb

```diff
@@ -6,17 +6,42 @@
 require "fileutils"
 
 describe "after Rails initialization" do
-  NAME = "#{File.dirname(__FILE__)}/../routes.js"
+  NAME = Rails.root.join('app', 'assets', 'javascripts', 'routes.js').to_s
 
-  before(:all) do
-    FileUtils.rm_f(NAME)
+  def sprockets_v3?
+    Sprockets::VERSION.to_i >= 3
+  end
+
+  def sprockets_context(environment, name, filename)
+    if sprockets_v3?
+      Sprockets::Context.new(environment: environment, name: name, filename: filename.to_s, metadata: {})
+    else
+      Sprockets::Context.new(environment, name, filename)
+    end
+  end
+
+  def evaluate(ctx, file)
+    if sprockets_v3?
+      ctx.load(ctx.environment.find_asset(file, pipeline: :default).uri).to_s
+    else
+      ctx.evaluate(file)
+    end
+  end
+
+  before(:each) do
+    FileUtils.rm_rf Rails.root.join('tmp/cache')
+    FileUtils.rm_f NAME
     JsRoutes.generate!(NAME)
-    Rails.configuration.eager_load = false
-    Rails.application.initialize!
   end
 
-  after(:all) do
-    FileUtils.rm_f(NAME)
+  after(:each) do
+    FileUtils.rm_rf Rails.root.join('tmp/cache')
+    FileUtils.rm_f NAME
```

Comment:
I think remove those files in `before` is enough. Why do you think `after` makes sense?